### PR TITLE
[Github Actions] Configure a workflow for running unit tests and static code analysis

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards #2670

Things done:
 * Triggers a github action on every pull request on master or push to master
 * Runs unit tests and static code analysis checks
 * Also enables gradle caching

@gino-m  PTAL?

### CODE REVIEW

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          distribution: 'temurin'
+          java-version: 17
+          cache: 'gradle'
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - name: Run unit tests
+        run: ./gradlew testDevDebugUnitTest --no-daemon
```
Comment (gino-m) [inline .github/workflows/unit-test.yml:41]: We already have a `localDebug` variant with a similar config. Perhaps we should use that one instead, and remove the duplicate local config in `google-services.json`?

**Review by gino-m — CHANGES_REQUESTED** [review]
That was quick! One comment.

File: .github/workflows/unit-test.yml (hunk 1 above)
Comment (shobhitagarwal1612) [reply .github/workflows/unit-test.yml]: Done

File: .github/workflows/unit-test.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+name: Unit Tests
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  test:
```
Comment (gino-m) [inline .github/workflows/unit-test.yml:23]: Can we split this into two jobs, `build` and `test`? `test` would just need `needs: build`.

File: .github/workflows/unit-test.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 The Ground Authors.
```
Comment (gino-m) [inline .github/workflows/unit-test.yml:1]: I assume this workflow will be used for all CI jobs? (build, test, lint, etc?) If so, should we rename it `ci.yml` or similar?

**Review by gino-m — CHANGES_REQUESTED** [review]
Naming nits incoming!

File: .github/workflows/unit-test.yml (hunk 3 above)
Comment (shobhitagarwal1612) [reply .github/workflows/unit-test.yml]: Done

**Comment by shobhitagarwal1612** [comment]
@gino-m Can you add the codecov token to project secrets? I don't have access to add it.

**Review by gino-m — APPROVED** [review]
🏅 !!

**Pull request merged**